File: findKthLargest/findKthLargest.cs
using System;

public partial class Solution
{
    public int FindKthLargest(int[] nums, int k)
    {
        var indexToFind = nums.Length - k;

        return QuickSelect(nums, 0, nums.Length - 1, indexToFind);
    }

    public void Swap(int[] array, int i, int j)
    {
        var temp = array[i];
        array[i] = array[j];
        array[j] = temp;
    }

    public int GetPartition(int[] nums, int left, int right)
    {
        var i = left;

        for (var j = left; j <= right; j++)
        {
            if (nums[j] <= nums[right])
            {
                Swap(nums, i, j);
                i++;
            }
        }
        return i - 1;
    }

    public int QuickSelect(int[] nums, int left, int right, int indexToFind)
    {
        var partitionIndex = GetPartition(nums, left, right);

        if (partitionIndex == indexToFind)
        {
            return nums[partitionIndex];
        }
        else if (indexToFind < partitionIndex)
        {
            return QuickSelect(nums, left, partitionIndex - 1, indexToFind);
        }
        else
        {
            return QuickSelect(nums, partitionIndex + 1, right, indexToFind);
        }
    }
}

File: findKthLargest/findKthLargest.py
def swap(array, i, j) :
  temp = array[i]
  array[i] = array[j]
  array[j] = temp


def getPartition(nums, left, right):
  i = left

  for j in range(left, right + 1):
    if  nums[j] <= nums[right] :
      swap(nums, i, j)
      i+=1
  return i - 1


def quickSort(nums, left, right):
  if left < right:
    partitionIndex = getPartition(nums, left, right)

    quickSort(nums, left, partitionIndex - 1)
    quickSort(nums, partitionIndex + 1, right)
  

def findKthLargest(nums, k) :
  indexToFind = len(nums) - k
  quickSort(nums, 0, len(nums) - 1)
  return nums[indexToFind]

array = [5,3,1,6,4,2]
kToFind = 4

print(findKthLargest(array, kToFind))
